src/libostree/ostree-core.c

```diff
@@ -830,22 +790,18 @@ ostree_checksum_file_from_input (GFileInfo        *file_info,
 
       if (!write_file_header_update_checksum (NULL, file_header, checksum,
                                               cancellable, error))
-        goto out;
+        return FALSE;
 
       if (g_file_info_get_file_type (file_info) == G_FILE_TYPE_REGULAR)
         {
           if (!ot_gio_splice_update_checksum (NULL, in, checksum, cancellable, error))
-            goto out;
+            return FALSE;
         }
     }
 
-  ret_csum = ot_csum_from_gchecksum (checksum);
-
-  ret = TRUE;
+  g_autofree guchar *ret_csum = ot_csum_from_gchecksum (checksum);
```

Comment:
In this case, yes.  As a general rule we have to be a bit careful since some functions have nullable out arguments.